### PR TITLE
feat: Android release process

### PR DESCRIPTION
**Pull request by Dmitry-Borodin**
## Purpose
Make release flow in github action instead of manual process in clean room.

## Scope
- build and sign APK in github
- upload release apk to internal track in play store
- readme updated of how we can release now.

Feedback is welcome.

### CODE REVIEW

**Review by montekki — APPROVED** [review]

File: .github/workflows/android-app.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ on:
       - master
       - stable
 jobs:
-  build:
-    name: Build
+  tests:
+    name: Build android unit tests
```
Comment (krodak) [inline .github/workflows/android-app.yml:11]: You need to revert to `Build` for name as this is required check for PR merge

**Review by krodak — CHANGES_REQUESTED** [review]
I think workflow trigger might be wrong and would trigger release on every push to main

**Comment by krodak** [comment]
My suggestion would be to align with iOS release pipeline and switch to manual trigger via CI:
```
on:
 workflow_dispatch:
```

File: .github/workflows/android-release.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
+name: Release android app
+
+on:
+  push:
+    branches:
+      - master
```
[the remark this answers is not in the document]
Comment (krodak) [reply .github/workflows/android-release.yml:6]: Here is additional extended command that suggest that indeed this is "OR" operation between branch name and tag. It also goes through suggestions on how to apply "AND" between those rules, using `if` triggers: https://stackoverflow.com/a/66463380

File: .github/workflows/android-release.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            ${{ runner.os }}-gradle-
+
+      # Build and publish
+
+      - name:           Assemble release unsigned apk
+        run:            |
+          ./gradlew clean assembleRelease --stacktrace
+
+      - name: Sign app APK
+        uses: r0adkll/sign-android-release@v1
```
Comment (prybalko) [inline .github/workflows/android-release.yml:145]: This repository seems abandoned. Could we use another one?
https://github.com/r0adkll/sign-android-release

File: .github/workflows/android-release.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          access_token: ${{ github.token }}
+
+      - name:           Checkout Sources
+        uses:           actions/checkout@v3.5.0
+        with:
+          fetch-depth:  50
+          submodules:   'recursive'
+
+      - name:           Install System Dependencies (Linux)
+        if: runner.os == 'Linux'
```
Comment (prybalko) [inline .github/workflows/android-release.yml:92]: What else could it be? Line 77 says `runs-on: ubuntu-22.04`

File: .github/workflows/android-release.yml [hunk 5]
```diff
@@ -0,0 +1,171 @@
+name: Release android app
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+jobs:
+  tests:
```
[the remark this answers is not in the document]
Comment (Dmitry-Borodin) [reply .github/workflows/android-release.yml:10]: I would keep it just to see just as a precaution that all tests passed for release task

[the remark this answers is not in the document]
Comment (krodak) [reply .github/workflows/android-release.yml]: Its pointless as with iOS release, as any code that is merged is going through that other workflow.
So you’d be calling same action twice on same code, thus increasing CI usage for no good reason

File: .github/workflows/android-release.yml (hunk 3 above)
Comment (prybalko) [reply .github/workflows/android-release.yml:141]: It uses deprecated Node12, so one day it'll stop working this summer
https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/
https://github.com/r0adkll/sign-android-release/pull/70

File: .github/workflows/android-release.yml (hunk 4 above)
Comment (Dmitry-Borodin) [reply .github/workflows/android-release.yml]: Good point, this part if copied from existing android flow to run tests. Removed.

**Comment by Dmitry-Borodin** [comment]
@krodak fixed issues you requested to change. Thx for review.

**Review by krodak — APPROVED** [review]
LGTM

File: .github/workflows/android-app.yml (hunk 1 above)
Comment (krodak) [reply .github/workflows/android-app.yml:12]: Judging by PR status I think it’s still outstanding @Dmitry-Borodin

**Pull request merged**